pkg/ccl/sqlccl/restore.go

```diff
@@ -772,7 +777,9 @@ func restore(
 		return failed, errors.Wrapf(err, "making import requests for %d backups", len(backupDescs))
 	}
 
-	restoreCtx, cancel := context.WithCancel(restoreCtx)
+	var cancel func()
+	restoreCtx, cancel = context.WithCancel(restoreCtx)
+	defer cancel()
```

Comment:
Was there a reason you needed to switch from shadowing `restoreCtx` to reassigning it?

Reply:
I'm not sure why this was legal before. The spec seems to say it shouldn't have worked.

> Unlike regular variable declarations, a short variable declaration may redeclare variables provided they were originally declared earlier in the same block (or the parameter lists if the block is the function body) with the same type, and at least one of the non-blank variables is new. As a consequence, redeclaration can only appear in a multi-variable short declaration. Redeclaration does not introduce a new variable; it just assigns a new value to the original.

https://golang.org/ref/spec#Short_variable_declarations